src/utils/isDerivedReactComponent.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+import { Component } from 'react'
+
+const cachedResults = new Map()
+
+export default function isDerivedReactComponent(fn: any) {
```

Comment:
What is this for?

If you want to check if function is a React component, you need to do `fn.prototype && fn.prototype.isReactComponent`. That's what React does.

Reply:
Ah okay, I'll change it in a followup PR. Thanks!

Reply:
You probably also won't want to cache it because I'd expect two property lookups to be faster than what this is doing

Reply:
Done, thanks for the feedback! https://github.com/styled-components/styled-components/pull/1946/files